consensus/istanbul/config.go

```diff
@@ -147,17 +149,35 @@ func (c Config) QBFTBlockNumber() int64 {
 
 // IsQBFTConsensusAt checks if qbft consensus is enabled for the block height identified by the given header
 func (c *Config) IsQBFTConsensusAt(blockNumber *big.Int) bool {
-	// If qbftBlock is not defined in genesis qbft consensus is not used
-	if c.TestQBFTBlock == nil {
-		return false
+	if c.TestQBFTBlock != nil {
+		if c.TestQBFTBlock.Uint64() == 0 {
+			return true
+		}
+
+		if blockNumber.Cmp(c.TestQBFTBlock) >= 0 {
+			return true
+		}
 	}
-
-	if c.TestQBFTBlock.Uint64() == 0 {
-		return true
+	for i := 0; c.Transitions != nil && i < len(c.Transitions) && c.Transitions[i].Block.Cmp(blockNumber) <= 0; i++ {
+		if c.Transitions[i].Algorithm == params.QBFT {
+			return true
+		}
 	}
+	return false
+}
 
-	if blockNumber.Cmp(c.TestQBFTBlock) >= 0 {
-		return true
+func (c Config) GetConfig(blockNumber *big.Int) Config {
+	newConfig := c
+	for i := 0; c.Transitions != nil && i < len(c.Transitions) && c.Transitions[i].Block.Cmp(blockNumber) <= 0; i++ {
```

Comment:
is c.Transitions ordered for sure?
https://go.dev/play/p/PX4bREJAiEg

Reply:
That is being verified at the init command by calling `CheckTransitionsData()` [code](https://github.com/ConsenSys/quorum/pull/1371/files#diff-10ff00a15ef58da7485fedeca3906c73cb236fde4b143a9d61601c63cbf1ffe8R613)